File: Second Semester(Java and C++)/Pascals Triangle(List 2)/Java/Main.java
public class Main {
    public static void main(String[] args) {
        if (args.length < 2) {
            System.out.println("Program requires at least 2 arguments, got " + args.length);
            return;
        }

        try { //Checking if the first argument(size of ArrayList) is correct
            final int n = Integer.parseInt(args[0]);
            final PascalsTriangleRowGenerator row = new PascalsTriangleRowGenerator(n);

            for (int i = 1; i < args.length; i++) {
                System.out.print(args[i] + " --> ");

                try {
                    final int elementIndex = Integer.parseInt(args[i]);
                    System.out.println(row.getElement(elementIndex));
                } 
                catch (final IllegalArgumentException e) {
                    System.out.println("not an integer");
                } 
                catch (final IndexOutOfBoundsException e) {
                    System.out.println(e.getMessage());
                }
            }
        }
        catch (final IllegalArgumentException e) {
            System.out.println("Program requires an integer in range <0, 34>, got " + args[0]);
        }
    }
}

File: Second Semester(Java and C++)/Pascals Triangle(List 2)/C++/main.cpp
#include "PascalsTriangleRowGenerator.hpp"
#include "PascalsTriangleRowMath.hpp"

#include <iostream>
#include <stdexcept>

int main(int argc, const char* const argv[]) {
    if (argc < 3) {
            std::cout << "Program requires at least 2 arguments, got " + (argc - 1) + '\n';
            return 0;
        }

        try { //Checking if the first argument(size of vector) is correct
            const int n = std::stoi(argv[1]);
            PascalsTriangleRowGenerator row(n);

            for (int i = 2; i < argc; i++) {
                std::cout << argv[i] << " --> ";

                try {
                    const int elementIndex = std::stoi(argv[i]);
                    std::cout << row.getElement(elementIndex);
                } 
                catch (const std::out_of_range &e) {
                    std::cout << e.what();
                }
                catch (const std::invalid_argument &e) {
                    std::cout << "not an integer";
                } 

                std::cout << '\n';
            }
        }
        catch (const std::invalid_argument &e) {
            std::cout << "Program requires an integer in range <0, 34>, got " << argv[1] << '\n';
        }
    
    return 0;
}
